keras.layers.Bidirectional constructor ignores trainable argument, causing models with trainable=False layers to incorrectly be cloned

mergian opened the pull request:
This PR fixes #17466.

qlzh727 commented on keras/layers/rnn/bidirectional_test.py:
```diff
@@ -1020,6 +1020,20 @@ def test_full_input_spec(self):
+        model = keras.Model(inp, bid(inp))
+
+        clone_trainable = keras.models.clone_model(model)
+        assert clone_trainable.get_config() == model.get_config()
```
Please use `self.assertEqual()` instead.

qlzh727 commented on keras/layers/rnn/bidirectional_test.py:
```diff
@@ -1020,6 +1020,20 @@ def test_full_input_spec(self):
+        bid.trainable = False
+
+        clone_untrainable = keras.models.clone_model(model)
+        assert clone_untrainable.get_config() == model.get_config()
```
Same here.

qlzh727 commented on keras/layers/rnn/bidirectional_test.py:
```diff
@@ -1020,6 +1020,20 @@ def test_full_input_spec(self):
+
+        clone_untrainable = keras.models.clone_model(model)
+        assert clone_untrainable.get_config() == model.get_config()
+
```
Can you also add a test to create a `keras.layers.Bidirectional` with a trainable = False in the __init__ arg?

qlzh727 reviewed (changes requested):
Thanks for the PR.

mergian commented:
I changed `self._trainable` to `= kwargs.get('trainable', layer.trainable)` as also other attributes get fetched from the layer. I also changed the test cases to test for these behavior.
